polar-core/src/formatting.rs

```diff
@@ -60,6 +60,7 @@ impl Trace {
 /// Traverse a [`Source`](../types/struct.Source.html) line by line until `offset` is reached,
 /// and return the source line containing the `offset` character as well as `num_lines` lines
 /// above and below it.
+// @TODO: Can we have the carrot under the whole range of the expression instead of just the beginning.
```

Comment:
```suggestion
// @TODO: Can we have the caret under the whole range of the expression instead of just the beginning.
```

🥕 

We should be able to, you can get the full span of a term with `term.span()`. Need to check what calls this though

Reply:
Yeah, lets have it be a followup. Think it's just stack traces and warnings other than this. Both seem like they'd also be improved by hi lighting the whole expression.